pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,429 @@
+        if x_axis in ["bottom", "top"] and y_axis in ["left", "right"]:
+            # use default plotitem if none provided
+            plotitem = self.pi
+        else:
+            # VIEW
+            plotitem = PlotItem(parent=self.pi, name=name, *args,
+                                enableMenu=False, **kwargs)  # pass axisitems?
+            # hide all plotitem axis (they vould interfere with viewbox)
+            for a in ["left", "bottom", "right", "top"]:
+                plotitem.hideAxis(a)
+            # fix parent legend not showing child charts
+            plotitem.legend = self.pi.legend
+            for k, pos, axis in [["left", [2, 0], y], ["bottom", [3, 1], x], ["right", [2, 2], y], ["top", [1, 1], x]]:
+                # # DO NOT USE, WILL MAKE AXIS UNMATCHED TO DATA
+                # # you can't add the new ones after, it doesn't work for some reason
+                # # hide them instead
+                # plotitem.layout.removeItem(plotitem.layout.itemAt(*pos))
+                if axis.orientation == k:
+                    plotitem.axes[k] = {"item": axis, "pos": pos}
+                # hide plotitem axes
+                plotitem.hideAxis(a)
+        # CHART
+        if chart is None:
+            chart = PlotDataItem(name=name)
+        plotitem.addItem(chart)
+        # keep plotitem inside chart
+        chart.plotItem = plotitem
+        # keeptrack of connections
+        if chart.name not in self._signalConnectionsByChart:
+            self._signalConnectionsByChart[chart.name] = {}
+        # keep axis track
+        chart.axes = [x_axis, y_axis]
+        # keep chart
+        self.charts[name] = chart
+        # create a mapping for this chart and his axis
+        self.axes[x_axis].charts.append(name)
+        self.axes[y_axis].charts.append(name)
+        return chart, plotitem
+
+    def clearLayout(self):
+        """Clears the widget but keeps all created axis and charts."""
+        while self.layout.count() > 0:
+            item = self.layout.itemAt(0)
+            self.layout.removeAt(0)
+            self.scene().removeItem(item)
+        for chart in self.charts.values():
+            self._disconnect_all(chart)
+
+    def makeLayout(self, axes=None, charts=None):
+        """Adds all given axes and charts to the widget.
+
+        Parameters
+        ----------
+        axes : list of str, optional
+            The names associated with the axes to show.
+        charts : list of PlotItems, optional
+            The names associated with the charts to show.
+        """
+        self.clearLayout()
+        _ = self._show_axes(axes)
+        shown_charts = self._show_charts(charts)
+        if len(shown_charts) != 0:
+            top_level = shown_charts[list(shown_charts)[-1]]
+            # FROM "https://github.com/pyqtgraph/pyqtgraph/pull/2010" by herodotus77
+            # propagate mouse actions to charts "hidden" behind
+            for k, c in self.charts.items():
```

Comment:
could also do `self.charts.values()` here since the key is not referenced

Reply:
done